.tekton/vector-pull-request.yaml

```diff
@@ -764,6 +764,80 @@ spec:
             operator: in
             values:
               - "false"
+      - name: sast-shellcheck
+        runAfter:
+          - build-container
+        when:
+          - input: $(params.skip-checks)
+            operator: in
+            values:
+              - "false"
+        params:
+        - name: image-digest
+          value: $(tasks.build-container.results.IMAGE_DIGEST)
+        - name: image-url
+          value: $(tasks.build-container.results.IMAGE_URL)
+        workspaces:
+          - name: workspace
+            workspace: workspace
+        taskSpec:
+          description: Scans source code for security vulnerabilities using ShellCheck, a Static Application Security Testing (SAST) tool.
+          params:
+            - description: Image URL.
+              name: image-url
+              type: string
+            - description: Image digest to report findings for.
+              name: image-digest
+              type: string
+          results:
+            - description: Tekton task test output.
+              name: TEST_OUTPUT
+          steps:
+            - image: quay.io/fedora/fedora:40
+              name: sast-shellcheck
+              script: |
+                #!/usr/bin/env bash
+                set -exuo pipefail
+
+                dnf install -y csdiff shellcheck
+
+                cd $(workspaces.workspace.path)/source
+
+                .ci/run-shellcheck.sh * > shellcheck-results.json
+                csgrep --mode=evtstat shellcheck-results.json
+                csgrep --mode=sarif shellcheck-results.json > $(workspaces.workspace.path)/hacbs/$(context.task.name)/shellcheck-results.sarif
+            - name: upload
+              image: quay.io/konflux-ci/oras:latest@sha256:04cbd5efbcf1a38944b050e3adbc3071218643f6aa92e95d5d25b173574bca5e
```

Comment:
This was initially cherry-picked from https://github.com/konflux-ci/build-definitions/pull/992 because I was experimenting with this before it was merged upstream.  Once you merge into into the main branch, the reference is kept updated by a Konflux bot, as for example here: 29fc71aed0fcda623080733a99530d86971ee001